systems/plants/RigidBodyManipulator.cpp

```diff
@@ -427,36 +442,71 @@ void RigidBodyManipulator::resize(int ndof, int num_featherstone_bodies, int num
 
 void RigidBodyManipulator::compile(void)
 {
+  /* todo:
+   - set joint limits (from drakejoint to rbm)
+   - add collision elements (and don't add them before, e.g. in constructModelmex).. or update them here
+   */
+
+  // reorder body list to make sure that parents before children in the list
+  for (int i=0; i<bodies.size(); i++) {
+    while (bodies[i]->hasParent()) {
+      auto iter = find(bodies.begin()+i+1,bodies.end(),bodies[i]->parent);
+      if (iter==bodies.end()) break;
```

Comment:
Wouldn't this mean that `bodies[i]->parent` isn't present in `bodies`? How could that happen? Should we throw an error here?